data-serving/reusable-data-service/reusable_data_service/stores/mongo_store.py

```diff
@@ -51,12 +51,22 @@ def count_cases(self, filter: Filter) -> int:
         return self.get_case_collection().count_documents(filter.to_mongo_query())
 
     def insert_case(self, case: Case):
-        to_insert = case.to_dict()
-        for field in Case.date_fields():
-            # BSON works with datetimes, not dates
-            to_insert[field] = date_to_datetime(to_insert[field])
+        to_insert = MongoStore.case_to_bson_compatible_dict(case)
         self.get_case_collection().insert_one(to_insert)
 
+    def batch_upsert(self, cases: List[Case]) -> Tuple[int, int]:
+        dicts = [MongoStore.case_to_bson_compatible_dict(c) for c in cases]
+        needs_insert = lambda d: d['caseReference'] is None or d['caseReference']['sourceId'] is None or d['caseReference']['sourceEntryId'] is None
```

Comment:
Should sourceId ever be None?

Reply:
In the original data source the whole caseReference can be optional and the sourceId and sourceEntryId are both optional fields, should I make that stricter here?

Reply:
It makes sense for sourceId to not be None, in the current line list I can't recall instances where that is not the case.